cli/cluster/cmd/replicaset.go

```diff
@@ -317,7 +314,66 @@ func AddRole(uri *url.URL, ctx RolesAddCtx) error {
 
 	source := replicaset.NewCConfigSource(collector, publisher,
 		replicaset.KeyPicker(pickPatchKey))
-	err = source.AddRole(replicaset.RolesAddCtx{
+	err = source.AddRole(replicaset.RolesUpdateCtx{
+		InstName:       ctx.InstName,
+		GroupName:      ctx.GroupName,
+		ReplicasetName: ctx.ReplicasetName,
+		IsGlobal:       ctx.IsGlobal,
+		RoleName:       ctx.RoleName,
+		Force:          ctx.Force,
+	})
+	if err == nil {
+		log.Info("Done.")
+	}
+	return err
+}
+
+// RolesRemoveCtx describes the context to remove role.
+type RolesRemoveCtx struct {
+	// InstName is an instance name in which remove role.
+	InstName string
+	// GroupName is an replicaset name in which remove role.
+	GroupName string
+	// ReplicasetName is an replicaset name in which remove role.
+	ReplicasetName string
+	// IsGlobal is an boolean value if role needs to remove in global scope.
+	IsGlobal bool
+	// RoleName is a name of role to remove.
+	RoleName string
+	// Publishers is data publisher factory.
+	Publishers libcluster.DataPublisherFactory
+	// Collectors is data collector factory.
+	Collectors libcluster.DataCollectorFactory
+	// Username defines a username for connection.
+	Username string
+	// Password defines a password for connection.
+	Password string
+	// Force true if the key selection for patching the config
+	// should be skipped.
+	Force bool
+}
+
+// RemoveRole removes a role by patching the cluster config.
+func RemoveRole(uri *url.URL, ctx RolesRemoveCtx) error {
```

Comment:
The same, it could be refactored to something common:
```suggestion
// RemoveRole removes a role by patching the cluster config.
func ChangeRole(uri *url.URL, ctx RolesChangeCtx, action RolesChanger/RolesChangeAction) error {
```
The `RolesChanger/RolesChangeAction` could be an interface/func type, or enum + switch.

Reply:
Refactored to the `RolesChanger` with func type.